src/transformers/tokenization_utils_base.py

```diff
@@ -3337,7 +3337,7 @@ def prepare_for_model(
 
         # Truncation: Handle max sequence length
         overflowing_tokens = []
-        if truncation_strategy != TruncationStrategy.DO_NOT_TRUNCATE and max_length and total_len > max_length:
+        if truncation_strategy != TruncationStrategy.DO_NOT_TRUNCATE and total_len > max_length:
```

Comment:
checking `and max_length` is required if `max_length` is `None`. Otherwise we will get an error comparing Nonetype and int.